### DISCUSSION TITLE
c/c++ configure

### DISCUSSION THREAD

**Discussion by JIMMYfenghe** [microsoft/vscode-cpptools#11355] [Q&A] [answered]
Regarding vscode in Ubuntu, enable c/c++ configuration and adopt the following configuration, but there are uninterpretable instructions in the standard library, such as typedef, and when using the system call open, I explicitly include the header file, but it cannot be In the case of jumping, it shows undefined, and I compiled and ran successfully, only the editor has a problem.
{
     "configurations": [
         {
             "name": "Linux",
             "includePath": [
                 "${workspaceFolder}/**",
                 "${workspaceFolder}/header"
             ],
             "defines": [],
             "compilerPath": "/usr/local/arm/gcc-linaro-4.9.4-2017.01-x86_64_arm-linux-gnueabihf/bin/arm-linux-gnueabihf-gcc",
             "cStandard": "c11",
             "cppStandard": "c++14",
             "intelliSenseMode": "gcc-arm"
         }
     ],
     "version": 4
}

**Comment by JIMMYfenghe** [comment, marked as answer]
The problem is solved. The root cause is the c/c++ version problem. Recently, I accidentally updated the extension to 1.17.x, which caused errors in identifying header files and instructions. Now my version is downgraded to 1.10.x and it is correct. No wavy lines appear, 1.10.x - 1.16.x not tried

**Reply by sean-mcmanus** [reply]
1.10.x is over a year old. It's recommended to downgrade to the previous version (i.e. 1.16.3) if the new version introduced a new issue.

**Reply by JIMMYfenghe** [reply]
However, there is also a problem in 16.x, and the same is true in 15x. Some header file sources cannot be opened, and 10.x is normal.

**Reply by JIMMYfenghe** [reply]
The above problems are based on browsing the linux kernel source code
